### DISCUSSION TITLE
Extrating origin, development names from the triangle

### DISCUSSION THREAD

**Discussion by genedan** [casact/chainladder-python#416] [General]
Suppose we initialized a triangle as follows:

```
cl.Triangle(
    data, origin='Acc Year', development=['Cal Year'], 
    columns=['PaidLoss'], cumulative=True)
```

Is it possible afterwards to extract the strings, "Acc Year" from the origin or "Cal Year" from the development of the triangle? Or, put it another way, are these strings stored somewhere in the triangle object? The reason why is I'm writing some functions to extract some of the triangle metadata which would include the period basis of the origin or development - but I won't have knowledge of these ahead of time.

For now, I've opted to default ("Accident " + origin_grain) for origin and ("Calendar" + development_grain) for development, and am only supporting this combination, but would like to cover possibilities such as policy year if the user supplies such a triangle. 

Then again, how likely am I to run in to these rarer cases? Most literature I run across has AY origin and CY development.

**Comment by jbogaardt** [comment]
We don't store any of the origin/development labeling from the original pandas dataframe. The column names can be anything. "Acc Year", "Column 1", "A", are all valid names. I don't see an issue in retaining this metadata, just haven't had a use case for it yet.

**Comment by genedan** [comment]
My use case happens after trying to get everything together in a table,  such as in the first column below, so far I've just stored the info elsewhere, but I think it would be really convenient if I could get it from a chainladder object:

![image](https://user-images.githubusercontent.com/6165650/222442859-290186d1-7f8c-469d-a902-8a3b677d819d.png)

I'll open a pull request shortly.

**Comment by jbogaardt** [comment]
I think there one piece we should probably deprecate to support this. Specifically, `Triangle` accepts a list of columns for `origin`, for example `['Year', 'Month']`.  Under the hood, it just concats them and converts to a datetime [here](https://github.com/casact/chainladder-python/blob/master/chainladder/core/base.py#L229-L231).  I'd guess <1% of chainladder users need this functionality, and I personally don't use it. However, it does make the column name you're after ambiguous.
